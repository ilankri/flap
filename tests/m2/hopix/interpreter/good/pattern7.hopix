type t = A | B (int) | C (int, int) | D (t, t)

val a = A
val b1 = B (1)
val c12 = C (1, 2)
val dab = D (a, b1)

val main =
  a ? {
  | A | B (_) =>
    print_string ("An A or a B!\n")
  | _ => print_string ("something else!\n")
  }
