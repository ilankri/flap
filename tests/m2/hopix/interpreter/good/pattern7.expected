An A or a B!
